components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/management/PreUpdatePasswordActionConverter.java

```diff
@@ -73,9 +74,13 @@ public Action buildAction(ActionDTO actionDTO) {
 
         Map<String, Object> properties = actionDTO.getProperties();
         PasswordSharing.Builder passwordSharingBuilder = new PasswordSharing.Builder();
-        passwordSharingBuilder.format((PasswordSharing.Format) actionDTO.getProperties().get(PASSWORD_SHARING_FORMAT));
+        if (properties.get(PASSWORD_SHARING_FORMAT) != null) {
+            passwordSharingBuilder.format((PasswordSharing.Format) ((ActionPropertyForService) actionDTO.getProperties()
+                    .get(PASSWORD_SHARING_FORMAT)).getValue());
+        }
```

Comment:
If we are doing this, let's throw a server error if the password sharing type is not present since it is a required field.
Keeping the null value in the object should not be the solution